app/src/main/java/edu/cmu/pocketsphinx/demo/PocketSphinxActivity.java

```diff
@@ -107,12 +121,38 @@ protected void onPostExecute(Exception result) {
     }
 
     @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+
+        if (requestCode == MY_PERMISSIONS_REQUEST_RECORD_AUDIO) {
```

Comment:
You can simplify this part in a similar way. I would also just extract runRecognizerSetup method and invoke it here instead of full activity restart.